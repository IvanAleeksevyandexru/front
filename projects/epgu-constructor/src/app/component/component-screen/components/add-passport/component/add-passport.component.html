<epgu-constructor-screen-pad class="screen-pad-body">
  <section class="add-passport-screen">
    <div class="add-passport-screen__body">
      <epgu-constructor-passport
        [attrs]="$any(data)"
      >
      </epgu-constructor-passport>
    </div>
  </section>
</epgu-constructor-screen-pad>
