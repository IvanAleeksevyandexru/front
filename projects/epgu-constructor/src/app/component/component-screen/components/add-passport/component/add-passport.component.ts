import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { ComponentBase } from '../../../../../screen/screen.types';

@Component({
  selector: 'epgu-constructor-add-passport',
  templateUrl: './add-passport.component.html',
  styleUrls: ['./add-passport.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AddPassportComponent {
  @Input() data: ComponentBase;
}
