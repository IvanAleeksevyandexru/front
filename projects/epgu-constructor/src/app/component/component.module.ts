import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { SharedModule } from '../shared/shared.module';


/**
 * Домен компанент. Сдесь храниться всё что связанно со компанентами и их обязками.
 *
 * TODO: Сюда будем переносить компаненты из домена скринов
 */
@NgModule({
  imports: [
    CommonModule,
    SharedModule
  ],
})
export class ComponentModule {}
