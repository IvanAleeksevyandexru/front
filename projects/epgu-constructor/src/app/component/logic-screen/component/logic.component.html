<div class="wrapper" *ngIf="isLoading$ | async">
  <lib-throbber-hexagon size="big"></lib-throbber-hexagon>
</div>
