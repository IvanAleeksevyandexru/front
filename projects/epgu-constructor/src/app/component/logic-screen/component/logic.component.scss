@import '~@epgu/epgu-constructor-ui-kit/src/styles/core';

.wrapper {
  position: fixed;
  top: 0;
  left: 0;
  min-width: 100%;
  min-height: 100%;
  z-index: 2147473647;
  opacity: 0.7;
  background: $color-white;
  display: flex;
  justify-content: center;
  align-items: center;
}
