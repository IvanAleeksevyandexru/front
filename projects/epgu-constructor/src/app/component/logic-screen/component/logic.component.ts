import { Component, ChangeDetectionStrategy } from '@angular/core';
import { catchError, delay, filter, map, switchMap, takeUntil, tap } from 'rxjs/operators';
import { BehaviorSubject, forkJoin } from 'rxjs';

import { ScreenService } from '../../../screen/screen.service';
import { ComponentValue } from '../logic.types';
import { LogicService } from '../service/logic.service';
import { UnsubscribeService } from '../../../core/services/unsubscribe/unsubscribe.service';

@Component({
  selector: 'epgu-constructor-logic',
  templateUrl: './logic.component.html',
  styleUrls: ['./logic.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  providers: [UnsubscribeService],
})
export class LogicComponent {
  isLoading$ = new BehaviorSubject<boolean>(false);
  components$ = this.screenService.logicComponents$.pipe(
    filter((components) => components.length > 0),
    map((components) => {
      this.isLoading$.next(true);
      return components.map((component) => ({
        id: component.id,
        value: JSON.parse(component.value) as ComponentValue,
      }));
    }),
    map((components) => this.logicService.fetch(components)),
    switchMap((components) => forkJoin(components)),
    map((response) => {
      return response.reduce(
        (acc, component) => ({
          ...acc,
          ...component,
        }),
        {},
      );
    }),
    tap((response) => {
      this.screenService.logicAnswers = response;
    }),
    catchError(() => this.screenService.logicComponents$),
    delay(500),
    tap(() => this.isLoading$.next(false)),
    takeUntil(this.unSubscribeService$),
  );

  constructor(
    private screenService: ScreenService,
    private logicService: LogicService,
    private unSubscribeService$: UnsubscribeService,
  ) {
    this.components$.subscribe();
  }
}
