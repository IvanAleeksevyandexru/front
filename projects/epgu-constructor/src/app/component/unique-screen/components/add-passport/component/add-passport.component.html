<epgu-constructor-screen-pad class="screen-pad-body" [formGroup]="form">
  <section class="add-passport-screen">
    <div class="add-passport-screen__body">
      <epgu-constructor-passport
        [attrs]="$any(attrs)"
        formControlName="passport"
        [suggestions]="(suggestions$ | async)[componentId$ | async]"
        (selectSuggest)="suggestHandle($event)">
      </epgu-constructor-passport>
    </div>
  </section>
</epgu-constructor-screen-pad>
