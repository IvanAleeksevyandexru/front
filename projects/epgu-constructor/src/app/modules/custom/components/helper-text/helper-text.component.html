<span class="helper-text">{{text}}</span>
