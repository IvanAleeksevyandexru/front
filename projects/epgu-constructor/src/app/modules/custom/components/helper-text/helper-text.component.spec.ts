import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { HelperTextComponent } from './helper-text.component';
import { By } from '@angular/platform-browser'

describe('HelperTextComponent', () => {
  let component: HelperTextComponent;
  let fixture: ComponentFixture<HelperTextComponent>;
  let helperTextMock = 'Awesome useful helper text.';

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ HelperTextComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HelperTextComponent);
    component = fixture.componentInstance;
    component.text = helperTextMock;
    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('component should render passed text', () => {
    const helperText = fixture.debugElement
      .query(By.css('.helper-text')).nativeElement.textContent;
    expect(helperText).toEqual(helperTextMock);
  });

});
