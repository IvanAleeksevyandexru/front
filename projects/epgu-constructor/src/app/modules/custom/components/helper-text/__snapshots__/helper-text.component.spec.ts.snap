// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelperTextComponent should match snapshot 1`] = `
<epgu-constructor-helper-text
  __ngContext__={[Function LRootView]}
  text={[Function String]}
>
  <span
    class="helper-text"
  >
    Awesome useful helper text.
  </span>
</epgu-constructor-helper-text>
`;
